albatross.ml

```diff
@@ -533,82 +552,127 @@ module Make (S : Tcpip.Stack.V4V6) = struct
     with
     | Error (`Msg msg) -> Error msg
     | Ok vmm_name ->
-        Result.map (fun c -> (vmm_name, c)) (gen_cert t ~domain cmd name)
+        Result.map (fun c -> (vmm_name, c)) (gen_cert state ~domain cmd name)
 
-  let query t ~domain ?(name = ".") ?push cmd =
-    match certs t domain name cmd with
+  let query stack state ~domain ?(name = ".") ?push cmd =
+    match certs state domain name cmd with
     | Error str -> Lwt.return (Error str)
-    | Ok (name, certificates) -> raw_query t ~name certificates cmd ?push reply
+    | Ok (vmm_name, certificates) ->
+        raw_query stack state.configuration ~name:vmm_name certificates cmd
+          ?push reply
 
-  let query_console t ~domain ~name f =
+  let query_console stack state ~domain ~name f =
     let cmd = `Console_cmd (`Console_subscribe (`Count 40)) in
-    match certs t domain name cmd with
+    match certs state domain name cmd with
     | Error str -> Lwt.return (Error str)
-    | Ok (name, certificates) ->
-        raw_query t ~name certificates cmd (console name f)
+    | Ok (vmm_name, certificates) ->
+        raw_query stack state.configuration ~name:vmm_name certificates cmd
+          (console vmm_name f)
 
-  let query_block_dump t ~domain ~name compression f =
+  let query_block_dump stack state ~domain ~name compression f =
     let cmd = `Block_cmd (`Block_dump compression) in
-    match certs t domain name cmd with
+    match certs state domain name cmd with
     | Error str -> Lwt.return (Error str)
-    | Ok (name, certificates) ->
-        raw_query t ~name certificates cmd (block_data name f)
+    | Ok (vmm_name, certificates) ->
+        raw_query stack state.configuration ~name:vmm_name certificates cmd
+          (block_data vmm_name f)
 
-  let set_policy t ~domain policy =
+  let set_policy stack state ~domain policy =
     let open Lwt.Infix in
     match Vmm_core.Name.path_of_string domain with
     | Error (`Msg msg) -> Lwt.return (Error ("couldn't set policy: " ^ msg))
     | Ok p -> (
-        (* we set it locally - which is then used for the next command *)
-        let old_policies = t.policies in
+        let old_policies = state.policies in
         let name = Vmm_core.Name.create_of_path p in
-        t.policies <- fst (Vmm_trie.insert name policy t.policies);
-        (* now we tell albatross about it, using a command for throwing it away *)
-        (* note that the 'certs' / 'gen_cert' uses the policies for intermediate certificates *)
-        query t ~domain (`Unikernel_cmd `Unikernel_info) >|= function
+        state.policies <- fst (Vmm_trie.insert name policy state.policies);
+        query stack state ~domain (`Unikernel_cmd `Unikernel_info) >|= function
         | Ok _ -> Ok (name, policy)
         | Error msg ->
             Logs.warn (fun m -> m "error updating policies: %s" msg);
-            t.policies <- old_policies;
+            state.policies <- old_policies;
             Error msg)
 end
```

Comment:
the changes are ok, I'm wondering why there's so much renaming going on. it is much easier to read the diff if you don't rename the variables & arguments.